src/server/routes/transaction/blockchain/getLogs.ts

```diff
@@ -0,0 +1,189 @@
+import { Static, Type } from "@sinclair/typebox";
+import { FastifyInstance } from "fastify";
+import { StatusCodes } from "http-status-codes";
+import {
+  eth_getTransactionReceipt,
+  getContract,
+  getRpcClient,
+  parseEventLogs,
+  prepareEvent,
+} from "thirdweb";
+import { resolveContractAbi } from "thirdweb/contract";
+import { getChain } from "../../../../utils/chain";
+import { thirdwebClient } from "../../../../utils/sdk";
+import { createCustomError } from "../../../middleware/error";
+import { standardResponseSchema } from "../../../schemas/sharedApiSchemas";
+import { getChainIdFromChain } from "../../../utils/chain";
+
+declare global {
+  interface BigInt {
+    toJSON: () => string;
+  }
+}
+BigInt.prototype.toJSON = function () {
+  return `${this}n`;
+};
+
+// INPUT
+const requestSchema = Type.Object({
+  txHash: Type.String({
+    description: "Transaction hash",
+    examples: [
+      "0xd9bcba8f5bc4ce5bf4d631b2a0144329c1df3b56ddb9fc64637ed3a4219dd087",
+    ],
+    pattern: "^0x([A-Fa-f0-9]{64})$",
+  }),
+  chain: Type.String({
+    examples: ["80002"],
+    description: "Chain ID or name",
+  }),
+});
+
+// OUTPUT
+export const responseBodySchema = Type.Object({
+  result: Type.Array(
+    Type.Object({
+      eventName: Type.String(),
+      args: Type.Unknown({
+        description:
+          "Event arguments. These are mostly key-value pairs, with values being numbers or strings. BigInts are represented as strings suffixed with 'n', eg: '1000000000000000000n'",
+        examples: [
+          {
+            from: "0xdeadbeeefdeadbeeefdeadbeeefdeadbeeefdead",
+            to: "0xdeadbeeefdeadbeeefdeadbeeefdeadbeeefdead",
+            value: "1000000000000000000n",
+          },
+        ],
+      }),
+      address: Type.String(),
+      topics: Type.Array(Type.String()),
+      data: Type.String(),
+      // this is a lie, we say bigint but we actually return a string
+      blockNumber: Type.BigInt({
```

Comment:
Why aren't we just returning a string here?

Reply:
typescript complains at the bottom, because technically we are returning a bigint from the function body